### PR TITLE
chore: bump babel to 7.16.x and update tests

### PR DESCRIPTION
**Pull request by armano2**
## Overview

Update babel package to 7.16 and align tests, this change is necessary for #4099 and #3147

closes #4085

### CODE REVIEW

File: packages/typescript-estree/tests/ast-alignment/utils.ts [hunk 1]
```diff
@@ -237,7 +237,7 @@ export function preprocessBabylonAST(ast: File): any {
       /**
        * Template strings seem to also be affected by the difference in opinion between different parsers in
        * @see https://github.com/babel/babel/issues/6681
-       * @see https://github.com/babel/babel/blob/main/eslint/babel-eslint-parser/src/convert/convertAST.js#L64-L80
+       * @see https://github.com/babel/babel/blob/main/eslint/babel-eslint-parser/src/convert/convertAST.cjs#L81-L102
```
Comment (JoshuaKGoldberg) [inline packages/typescript-estree/tests/ast-alignment/utils.ts:240]: While you're here, it'd probably be better to make this a longer lasting permalink.

```suggestion
       * @see https://github.com/babel/babel/blob/381277a/eslint/babel-eslint-parser/src/convert/convertAST.cjs#L81-L102
```

File: packages/typescript-estree/tests/lib/semanticInfo-singleRun.test.ts [hunk 2]
```diff
@@ -195,7 +195,7 @@ describe('semanticInfo - singleRun', () => {
      * Single run because of process.argv
      */
     const originalProcessArgv = process.argv;
-    process.argv = ['', 'node_modules/.bin/eslint', ''];
+    process.argv = ['', path.normalize('node_modules/.bin/eslint'), ''];
```
Comment (JoshuaKGoldberg) [inline packages/typescript-estree/tests/lib/semanticInfo-singleRun.test.ts:198]: Heh, is this for Windows? If so, thanks! 😄

**Review by JoshuaKGoldberg — APPROVED** [review]
LGTM, thanks!

File: packages/typescript-estree/tests/lib/semanticInfo-singleRun.test.ts (hunk 2 above)
Comment (armano2) [reply packages/typescript-estree/tests/lib/semanticInfo-singleRun.test.ts:198]: yes it is :)

**Pull request merged**